src/compnents/PopupSignin/PopupSignin.js

```diff
@@ -1,13 +1,32 @@
-import React from 'react';
-import PopupWithForm from '../PopupWithForm/PopupWithForm'
+import React, { useState } from 'react';
+import { useFormWithValidation } from '../../hooks/useFormWithValidation';
+import PopupWithForm from '../PopupWithForm/PopupWithForm';
 
 const PopupSignin = (props) => {
   const { isOpen, onClose, onSubmit, switchPopups } = props;
+    const { values, handleChange, errors, isValid, resetForm } = useFormWithValidation();
+
+    // const [values, setValues] = useState({});
+    // const handleChange = (event) => {
+    //   setValues({ ...values, [event.target.name]: event.target.value });
+    // };
+
+    const handleSubmit = (event) => {
+      event.preventDefault();
+      if (isValid) {
+        onSubmit({ email: values.email, password: values.password });
+        resetForm();
```

Comment:
`        resetForm();
`

You need to **reset** the form only when you **open** it not to do that immediately after `submit` because when there is an error the popup will be opened and the user can try to send the data again to the server without having to print the data into the inputs once more



```js
    React.useEffect(() => {
        resetForm();
    }, [isOpen]);

```